From d68e177bdb9c51cf05dafa644eec8889824e661b Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:59:36 -0400
Subject: [ethernet] mlx4: slight optimization of addr compare

Message-id: <52fbc2862b40de7a0029caea36bd109675bb0e00.1410425016.git.avadai@redhat.com>
Patchwork-id: 93622
O-Subject: [Patch RHEL7.1 V1 040/197] net: mlx4: slight optimization of addr compare
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: dingtianhong <dingtianhong@huawei.com>

Use possibly more efficient ether_addr_equal
to instead of memcmp.

Cc: Amir Vadai <amirv@mellanox.com>
Signed-off-by: Ding Tianhong <dingtianhong@huawei.com>
Acked-By: Amir Vadai <amirv@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit c0623e587d869b4b18e077d64a8524ea364b5b77)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/en_netdev.c b/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
index b68dde0..60b927e 100644
--- a/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
+++ b/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
@@ -841,7 +841,7 @@ static void update_mclist_flags(struct mlx4_en_priv *priv,
  list_for_each_entry(dst_tmp, dst, list) {
   found = false;
   list_for_each_entry(src_tmp, src, list) {
-   if (!memcmp(dst_tmp->addr, src_tmp->addr, ETH_ALEN)) {
+   if (ether_addr_equal(dst_tmp->addr, src_tmp->addr)) {
     found = true;
     break;
    }
@@ -856,7 +856,7 @@ static void update_mclist_flags(struct mlx4_en_priv *priv,
  list_for_each_entry(src_tmp, src, list) {
   found = false;
   list_for_each_entry(dst_tmp, dst, list) {
-   if (!memcmp(dst_tmp->addr, src_tmp->addr, ETH_ALEN)) {
+   if (ether_addr_equal(dst_tmp->addr, src_tmp->addr)) {
     dst_tmp->action = MCLIST_NONE;
     found = true;
     break;
diff --git a/drivers/net/ethernet/mellanox/mlx4/resource_tracker.c b/drivers/net/ethernet/mellanox/mlx4/resource_tracker.c
index 03a213d..57428a0 100644
--- a/drivers/net/ethernet/mellanox/mlx4/resource_tracker.c
+++ b/drivers/net/ethernet/mellanox/mlx4/resource_tracker.c
@@ -3606,7 +3606,7 @@ static int validate_eth_header_mac(int slave, struct _rule_hw *eth_header,
      !is_broadcast_ether_addr(eth_header->eth.dst_mac)) {
   list_for_each_entry_safe(res, tmp, rlist, list) {
    be_mac = cpu_to_be64(res->mac << 16);
-   if (!memcmp(&be_mac, eth_header->eth.dst_mac, ETH_ALEN))
+   if (ether_addr_equal((u8 *)&be_mac, eth_header->eth.dst_mac))
     return 0;
   }
   pr_err("MAC %pM doesn't belong to VF %d, Steering rule rejected\n",
-- 
1.7.1